### DISCUSSION TITLE
[@types/lodash] Argument of type 'number[][]' is not assignable to parameter of type 'ArrayLike<number[][]>'

### DISCUSSION THREAD

**Discussion by artidataio** [DefinitelyTyped/DefinitelyTyped#58858] [Issues with a @types package] [answered]
In Javascript project the following code will do just fine:
```js
console.log(_.xor([[2,2],[1,1]],[[1,1]])) //[[2,2]]
```
but it will report error:

```error
Argument of type 'number[][]' is not assignable to parameter of type 'ArrayLike<number[][]>'
```
What should the correction be?

[a comment by an automated account is not shown]

**Comment by aj-r** [comment, marked as answer]
What are you trying to accomplish? When I run that code I get a different result: `[[2,2], [1,1], [1,1]]`. It seems xor doesn't work properly with nested arrays. It's hard to know what the correction should be without knowing what you're trying to do.

**Reply by artidataio** [reply]
Sorry for not being clear and erroneous. I'm trying to convert a react project where all the js files to be changed to ts files.  There's a line in the codebase:

```js
_.differenceWith(
    [[2, 2],[1, 1]],
    [[1, 1]],
    _.isEqual
  )
// returned [2,2]
```

When turning it into a ts file, it fails to run and report error:
```js
// Argument of type 'number[][]' is not assignable to parameter of type 'ArrayLike<number[][]>'
```

What's your suggestion to prevent the error?

You're right and I was wrong about: 

> When I run that code I get a different result: `[[2,2], [1,1], [1,1]]`

I didn't check the line I reported earlier. As I was planning to refactor to that too. I was assuming that it will work as other example suggest.

**Reply by aj-r** [reply]
Thanks for the info. For me that code compiles successfully, as does this:

```ts
_.xorWith(
    [[2, 2],[1, 1]],
    [[1, 1]],
    _.isEqual
  )
```

I'm using `typescript@4.5.5` and `@types/lodash@4.14.178`.

Is it possible that the error is somewhere else?
